clippy_lints/src/lib.rs

```diff
@@ -126,11 +126,16 @@ macro_rules! declare_clippy_lint {
             $(#[$attr])* pub clippy::$name, Allow, $description, report_in_external_macro: true
         }
     };
-    { $(#[$attr:meta])* pub $name:tt, restriction, $description:tt } => {
+
+    { $(#[$attr:meta])* pub $name:ident, restriction, $description:tt } => {
+        declare_clippy_lint! { $(#[$attr])* pub $name, restriction, $description, level = Allow }
+    };
+    { $(#[$attr:meta])* pub $name:ident, restriction, $description:tt, level = $level:ident } => {
```

Comment:
This changed from `tt` to `ident` because it made the error message way easier to read. Before it was

```
error: no rules expected the token `LET_UNDERSCORE_MUST_USE`
  --> clippy_lints/src/let_underscore.rs:30:9
   |
30 |     pub LET_UNDERSCORE_MUST_USE,
   |         ^^^^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
   | 
  ::: clippy_lints/src/lib.rs:98:1
   |
98 | macro_rules! declare_clippy_lint {
   | -------------------------------- when calling this macro

```

now it's
```
error: no rules expected the token `LET_UNDERSCORE_MUST_USE`
   --> clippy_lints/src/lib.rs:131:45
    |
98  |   macro_rules! declare_clippy_lint {
    |   -------------------------------- when calling this macro
...
131 |           declare_clippy_lint! { $(#[$attr])* $name, restriction, $description, level = Allow }
    |                                               ^^^^^ no rules expected this token in macro call
    | 
   ::: clippy_lints/src/let_underscore.rs:11:1
    |
11  | / declare_clippy_lint! {
12  | |     /// **What it does:** Checks for `let _ = <expr>`
13  | |     /// where expr is #[must_use]
14  | |     ///
...   |
32  | |     "non-binding let on a `#[must_use]` expression"
33  | | }
    | |_- in this macro invocation
    |
    = note: this error originates in the macro `declare_clippy_lint` (in Nightly builds, run with -Z macro-backtrace for more info)
```

(the original issue was I forgot `pub` in front of `clippy::$name` below)